opencog/atoms/core/ScopeLink.cc

```diff
@@ -313,29 +358,23 @@ ContentHash ScopeLink::term_hash(const Handle& h,
 		const Variables& vees = sco->get_variables();
 		for (const Handle& v : vees.varseq) bound_vars.insert(v);
 	}
-
-	// Prevent mixing for UnorderedLinks. The `mixer` var will be zero
-	// for UnorderedLinks. The problem is that two UnorderdLinks might
-	// be alpha-equivalent, but have their atoms presented in a
-	// different order. Thus, the hash must be computed in a purely
-	// commutative fashion: using only addition, so as to never create
-	// any entropy, until the end.
-	//
-	// XXX As discussed in issue #1176, a better fix would be to
-	// compute the individual term_hashes first, then sort them,
-	// and then mix them!  This provides the desired qualities:
-	// different unordered links can be directly compared, and also
-	// have good mixing/avalanching properties. The code below
-	// only allows for compare; it fails to mix.
-	//
-	bool is_ordered = not classserver().isA(t, UNORDERED_LINK);
-	ContentHash mixer = (ContentHash) is_ordered;
+	// As discussed in issue #1176, compute the individual term_hashes
```

Comment:
Don't use github issues as replacements for actual documentation. Make the documentation sufficiently clear first, and then, for more info, refer to actual perma-URL's. Don't refer to issue #'s because its not clear which repo that issue is from  (we've got maybe 50 or 100 repos now).

Reply:
Also, deletion of the comment above is... dangerous: it says something important about how unordered links need to be handled: they behave very differently than ordered links.  Loosing track of this will create subtle, confusing bugs.

Reply:
Specifically, its not clear to me that you understood the comment about alpha-equivalence and commutativity.  Specifically,
```
    Lambda A B
          Set A B
```
is alpha-equivalent to 
```
Lambda  P Q
     Set P Q
```
but the UnordererdLink for `(Set A B)` and the unorderedLink for `(Set P Q)` might place the contents in a different order.  Thus, if you use mixing during the hash computations, you will get two different hashes, when, in fact, you should have gotten exactly the same hash for both cases, viz, independent of the order in the unordered link.

I mean, I would need to look at this much more closely, to be convinced that you are doing the right thing.  Are you SURE you are doing the right thing?

Reply:
I tested and it works (using different atomspaces to be sure not to enable automatic alpha-conversion). This is consistent with my understanding of the code since only considers the variable indexes as opposed to their names (you wrote that code).